lib/trino-parquet/src/test/java/io/trino/parquet/writer/TestParquetWriter.java

```diff
@@ -152,7 +153,9 @@ public void testLargeStringTruncation()
         ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
         BlockMetaData blockMetaData = getOnlyElement(parquetMetadata.getBlocks());
         ColumnChunkMetaData chunkMetaData = getOnlyElement(blockMetaData.getColumns());
-        assertThat(chunkMetaData.getStatistics().isEmpty()).isTrue();
+        assertThat(chunkMetaData.getStatistics().getMinBytes()).isEqualTo(min.getBytes());
+        Slice truncatedMax = Slices.utf8Slice("y".repeat(1023) + "z");
```

Comment:
let's have test coverage for incrementing last character when one or more last characters are "max codepoint". see `io.trino.type.TestCharacterStringCasts#testVarcharToCharSaturatedFloorCast` for good copy-pasting source

let's also have test coverage for case where truncation would chop a character in the middle of its bytes

Reply:
going to add these tests, just wanted to validate the rest of the approach first

Reply:
understood, thanks